crates/wasmtime/src/compile/code_builder.rs

```diff
@@ -76,6 +80,10 @@ impl<'a> CodeBuilder<'a> {
         }
         self.wasm = Some(wasm_bytes.into());
         self.wasm_path = wasm_path.map(|p| p.into());
+        if self.wasm_path.is_some() {
+            self.dwarf_package_from_wasm_path()?;
+        }
```

Comment:
If it works for you I'd prefer to leave out this in this method if we can. I think it's good to have some "pure in memory methods" for the builder that don't touch the filesystem, and this `wasm` method was intended for that. The `wasm_file` is a nice convenience for probing the filesystem for `*.dwp`, but here `wasm_path` is mostly an error message hint rather than a location on disk per se.

Reply:
sure, removed.